conceptual/Npgsql/release-notes/7.0.md

```diff
@@ -2,14 +2,38 @@
 
 Npgsql version 7.0 has been released and is available on nuget.
 
-Npgsql 6.0 brings some major breaking changes and is not a simple in-place upgrade. Carefully read the breaking change notes below and upgrade with care.
-
 ## New features
 
 The full list of issues for this release is [available here](https://github.com/npgsql/npgsql/milestone/71?closed=1).
 
 ### DbDataSource
 
+A major improvement in Npgsql 7.0 is <xref:Npgsql.NpgsqlDataSource>, which implements the new [DbDataSource](https://learn.microsoft.com/dotnet/api/system.data.common.dbdatasource?view=net-7.0) abstraction in .NET System.Data. A data source represents your PostgreSQL database, and can hand out connections to it or support direct execution of SQL to it.
+
+Instead of directly instantiating an `NpgsqlConnection` and then executing commands against it, you now create a data source once, and then use that throughout your application:
+
+```csharp
+await using var dataSource = NpgsqlDataSource.Create(connectionString);
+
+// Execute a command directly against the data source, no NpgsqlConnection needed:
+await using var command = dataSource.CreateCommand("INSERT INTO some_table (some_field) VALUES (8)");
+await command.ExecuteNonQueryAsync();
+
+// Open a connection in order to e.g. start a transaction on it:
+await using var connection = await dataSource.OpenConnectionAsync();
+```
+
+Since the data source encapsulates all the necessary configuration for connecting to a database (e.g. the connection string, authentication callbacks...), it can registered in dependency injection or passed around as needed, without needing any additional information. The new <xref:Npgsql.NpgsqlDataSourceBuilder> also  provides the ideal API point for various configuration when building a data source:
+
+```csharp
+var dataSourceBuilder = new NpgsqlDataSourceBuilder("Host=localhost;Username=test;Password=test");
+dataSourceBuilder
+    .UseLoggerFactory(loggerFactory) // Configure logging
+    .UseNodaTime() // Use NodaTime for date/time types
+    .UsePeriodicPasswordProvider(); // Automatically rotate the password periodically
```

Comment:
This doesn't actually work because `UseNodaTime()` returns a `INpgsqlTypeMapper` not the `NpgsqlDataSourceBuilder` to allow chaining.

I've raised npgsql/npgsql#4751 to fix this, if you'd like to take a look - it's pretty trivial.